File: Mbs.Api/Models/Trading/Instruments/Stock.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Runtime.Serialization;
using Mbs.Trading.Currencies;

namespace Mbs.Api.Models.Trading.Instruments
{
    /// <summary>
    /// An additional information for stocks.
    /// </summary>
    [DataContract]
    public class Stock : IValidatableObject
    {
        /// <summary>
        /// Gets or sets a currency code.
        /// </summary>
        [DataMember(IsRequired = false, EmitDefaultValue = false)]
        public CurrencyCode Currency { get; set; } = CurrencyCode.Xxx;

        /// <summary>
        /// Gets or sets a trading mode.
        /// </summary>
        [DataMember(IsRequired = false, EmitDefaultValue = false)]
        public string TradingMode { get; set; }

        /// <summary>
        /// Gets or sets an ISO 10962 Classification of Financial Instruments code.
        /// </summary>
        [DataMember(IsRequired = false, EmitDefaultValue = false)]
        public string Cfi { get; set; }

        /// <summary>
        /// Gets or sets an Industry Classification Benchmark code.
        /// </summary>
        [DataMember(IsRequired = false, EmitDefaultValue = false)]
        public string Icb { get; set; }

        /// <summary>
        /// Gets or sets a number of shares outstanding.
        /// </summary>
        [DataMember(IsRequired = false, EmitDefaultValue = false)]
        public long SharesOutstanding { get; set; }

        /// <inheritdoc />
        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
        {
            yield break;
        }
    }
}

File: Mbs.Api.Host.Ng/ClientApp/src/app/shared/mbs/instruments/types/stock.ts
import { CurrencyCode } from '../../currencies/currency-code.enum';

/** An additional information for stocks. */
export class Stock {
  /** A currency code. */
  currency!: CurrencyCode;
  /** A trading mode. */
  tradingMode?: string;
  /** An ISO 10962 *Classification of Financial Instruments* code. */
  cfi?: string;
  /** An Industry Classification Benchmark code. */
  icb?: string;
  /** A number of shares outstanding. */
  sharesOutstanding?: number | undefined;

  constructor(data?: Stock) {
    if (data) {
      for (const property in data) {
        if (data.hasOwnProperty(property)) {
          (this as any)[property] = (data as any)[property];
        }
      }
    }
  }

  toJSON(data?: any) {
    data = typeof data === 'object' ? data : {};
    // tslint:disable:no-string-literal
    data['Currency'] = this.currency;
    data['TradingMode'] = this.tradingMode;
    data['Cfi'] = this.cfi;
    data['Icb'] = this.icb;
    data['SharesOutstanding'] = this.sharesOutstanding;
    // tslint:enable:no-string-literal
    return data;
  }
}
